python/src/dataset.rs

```diff
@@ -275,39 +278,72 @@ impl Dataset {
         let idx_type = match index_type.to_uppercase().as_str() {
-            "IVF_PQ" => IndexType::Vector,
+            "IVF_PQ" | "DISKANN" => IndexType::Vector,
             _ => {
                 return Err(PyValueError::new_err(format!(
                     "Index type '{index_type}' is not supported."
                 )))
             }
         };
 
-        // Only VectorParams are supported.
-        let mut params = VectorIndexParams::default();
-        if let Some(kwargs) = kwargs {
-            if let Some(n) = kwargs.get_item("num_partitions") {
-                params.num_partitions = PyAny::downcast::<PyInt>(n)?.extract()?
-            };
-
-            if let Some(n) = kwargs.get_item("num_sub_vectors") {
-                params.num_sub_vectors = PyAny::downcast::<PyInt>(n)?.extract()?
-            };
-
-            if let Some(o) = kwargs.get_item("use_opq") {
-                params.use_opq = PyAny::downcast::<PyBool>(o)?.extract()?
-            };
-            if let Some(o) = kwargs.get_item("max_opq_iterations") {
-                params.max_opq_iterations = PyAny::downcast::<PyInt>(o)?.extract()?
-            };
-        }
-
-        params.metric_type = match metric_type {
+        let m_type = match metric_type {
             Some(mt) => MetricType::try_from(mt.to_string().to_lowercase().as_str())
                 .map_err(|err| PyValueError::new_err(err.to_string()))?,
             None => MetricType::L2,
         };
 
+        // Only VectorParams are supported.
+        let params = match index_type.to_uppercase().as_str() {
+            "IVF_PQ" => {
+                let mut ivf_params = IvfBuildParams::default();
+                let mut pq_params = PQBuildParams::default();
+                if let Some(kwargs) = kwargs {
+                    if let Some(n) = kwargs.get_item("num_partitions") {
+                        ivf_params.num_partitions = PyAny::downcast::<PyInt>(n)?.extract()?
+                    };
+
+                    if let Some(n) = kwargs.get_item("num_bits") {
+                        pq_params.num_bits = PyAny::downcast::<PyInt>(n)?.extract()?
+                    };
+
+                    if let Some(n) = kwargs.get_item("num_sub_vectors") {
+                        pq_params.num_sub_vectors = PyAny::downcast::<PyInt>(n)?.extract()?
+                    };
+
+                    if let Some(o) = kwargs.get_item("use_opq") {
+                        pq_params.use_opq = PyAny::downcast::<PyBool>(o)?.extract()?
+                    };
+
+                    if let Some(o) = kwargs.get_item("max_opq_iterations") {
+                        pq_params.max_opq_iters = PyAny::downcast::<PyInt>(o)?.extract()?
+                    };
+                }
+                VectorIndexParams::with_ivf_pq_params(m_type, ivf_params, pq_params)
+            }
+            "DISKANN" => {
+                let mut params = DiskANNParams::default();
+                if let Some(kwargs) = kwargs {
```

Comment:
what're the default values for these? prolly worth tuning it and set reasonable defaults for like high embedding dimensions and like up to 1M vectors ?

Reply:
The default values were from the diskann paper. Will adjust them if necessary when we benchmark diskann